cranelift/codegen/src/machinst/buffer.rs

```diff
@@ -1467,6 +1467,59 @@ impl<T: CompilePhase> MachBufferFinalized<T> {
     pub fn call_sites(&self) -> &[MachCallSite] {
         &self.call_sites[..]
     }
+
+    /// Get the disassembly of the buffer, using the given capstone context.
+    #[cfg(feature = "disas")]
+    pub fn disassemble(
+        &self,
+        params: Option<&crate::ir::function::FunctionParameters>,
+        cs: &capstone::Capstone,
+    ) -> Result<String, anyhow::Error> {
+        use std::fmt::Write;
+
+        let mut buf = String::new();
+
+        let relocs = self.relocs.as_slice();
+        let traps = self.traps.as_slice();
+
+        let insns = cs.disasm_all(&self.data, 0x0).map_err(map_caperr)?;
+        for i in insns.iter() {
+            write!(buf, "  ")?;
+
+            if let Some(s) = i.mnemonic() {
+                write!(buf, "{}\t", s)?;
+            }
+
+            if let Some(s) = i.op_str() {
+                write!(buf, "{}", s)?;
+            }
+
+            let end = i.address() + i.bytes().len() as u64;
```

Comment:
If I'm reading the documentation correctly you can just ask the instruction for its length, you don't need to get its bytes first.

```suggestion
            let end = i.address() + i.len() as u64;
```

Reply:
I tried this and got a compilation error as `len` is private. Perhaps it's public in a newer version of `capstone`?

Reply:
Ah, I was looking at the docs for 0.11 and we're using 0.9, which indeed didn't have it. Nevermind!